flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/rules/logical/RemoveUnreachableCoalesceArgumentsRuleTest.java

```diff
@@ -83,4 +83,27 @@ public void testJoinCoalesce() {
         util.verifyRelPlan(
                 "SELECT * FROM T t1 LEFT JOIN T t2 ON COALESCE(t1.f0, '-', t1.f2) = t2.f0");
     }
+
+    @Test
+    public void testMultipleCoalesces() {
```

Comment:
This test is not super helpful as it is unable to test to final result type. Can you add at least a single test to `CoalesceFunctionITCase`?